src/main.cpp

```diff
@@ -2636,9 +2636,10 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
             // Notify external listeners about the new tip.
             if (!vHashes.empty()) {
                 GetMainSignals().UpdatedBlockTip(pindexNewTip);
-                uiInterface.NotifyBlockTip(vHashes.front());
             }
         }
+        // Always notify the UI if a new block tip was connected
+        uiInterface.NotifyBlockTip(fInitialDownload, pindexNewTip);
```

Comment:
No, you should only fire if vHashes is not empty. vHashes contains the
block hashes of blocks that were not in the best chain, but now are.